R/nneo_wrangle.R

```diff
@@ -0,0 +1,108 @@
+#############################################################################################
+#' @title Download and wrangle NEON's sensor-based data products
+
+#' @author Josh Roberti \email{jaroberti87@gmail.com}\cr
+#' Dave Durden\cr
+#' Robert Lee
+
+#' @description Retrieve related, sensor-based datasets at a given NEON site for a custom time period, merged per measurement level and/or variable
+#'
+#' @param site_code (character) a site code. Required.
+#' @param time_start (character) YYYY-MM-DD start day to check for files. Required if \code{is.null(time_end)}
+#' @param time_end (character) YYYY-MM-DD end day to check for files. Required if \code{is.null(time_start)}
+#' @param time_agr (numeric) Aggregation period, given in minutes, of NEON data product(s) to be downloaded (e.g. `1` corresponds to a 1-minute data product). Required.
+#' @param data_var (character) NEON data product(s) to be downloaded. The user may enter a specific, controlled, NEON data product or they may enter a generic term if wanting multiple, similar, NEON data products. For instance, defining \code{data_var} as "Photosythetically Active Radiation (Quantum Line)" would satisfy the former, and defining \code{data_var} as "Active Radiation" or just "Radiation" would satisfy the latter. Required
+#' @param package (character) Package type to return, basic or expanded. 'Expanded' datasets are only available for the smallest \code{time_agr} of each NEON data product. NOTE: 'expanded' datasets are much larger than 'basic' and will take considerably more time to download. Required
+
+#' @return Returns a data.frame of relevant data products from all spatial locations at a NEON site for the custom time period.  Data products are displayed via 'productName.spatialLocation', e.g., 'difRadMean.000.060' is mean, diffuse, shortwave radiation as measured on the 6th measurement level of a NEON tower (000.060), while 'linePARMean.005.000' is mean, photosynthetically Active Radiation (PAR) as measured at the 5th soil plot (005.000) of a NEON site.  More information regarding spatial location and identifiers can be found at the references below.
+#'
+#' @references
+#' NEON Data Portal \url{http://data.neonscience.org/home}\cr
+#' NEON Data Product Availability \url{http://data.neonscience.org/view-data-availability}
+
+#' @keywords Ecology, environmental data, climate, data, data products,
+#' National Ecological Observatory Network (NEON), NEON, atmosphere
+
+#' @examples
```

Comment:
i'd wrap your examples in `dontrun{}` because we don't want examples running on CRAN that can fail, then they get mad at us